File: test/performance/udpserver.cc
// g++ -std=c++11 -O3 -Wall -Wextra -o udpserver udpserver.cc -lPocoNet

#include <thread>
#include <iostream>
#include <atomic>

#include <Poco/Net/DatagramSocket.h>
#include <Poco/Net/SocketAddress.h>

std::atomic_ulong packets_received(0);

void count_packets() {
   while(true) {
      unsigned long const opr(packets_received.load());
      sleep(1);
      unsigned long const delta_packets_received
         = packets_received.load() - opr;
      std::cout << packets_received.load() << " "
                << delta_packets_received << std::endl;
   }
}

int main() {

   std::thread t(count_packets);

   Poco::Net::SocketAddress sa("127.0.0.1", 8125);
   Poco::Net::DatagramSocket dgs(sa);

   char buf[1024];

   while(true) {
      dgs.receiveBytes(buf, 1024);
      ++packets_received;
   }

   return 0;
}

File: test/performance/udpserver.py
import socket
import time
import thread

UDP_IP = "127.0.0.1"
UDP_PORT = 8125

sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
sock.bind((UDP_IP, UDP_PORT))

packets_received = 0

def count_packets():
    while True:
        opr = packets_received
        time.sleep(1.0)
        delta_packets_received = packets_received - opr
        print("%7d %7d" % (packets_received, delta_packets_received))

thread.start_new_thread(count_packets, ())

while True:
    data, addr = sock.recvfrom(1024)
    packets_received += 1
